src/full/Agda/TypeChecking/Serialise/Base.hs

```diff
@@ -67,13 +69,23 @@ instance Hashable Node where
     foldedMul (W# x) (W# y) = case timesWord2# x y of (# hi, lo #) -> W# (xor# hi lo)
 
     combine :: Word -> Word -> Word
-    combine x y = foldedMul (xor x y) 11400714819323198549
+    combine x y = foldedMul (xor x y) factor where
+#if WORD_SIZE_IN_BITS == 64
+    factor = 11400714819323198549
+#else
+    factor = 2654435741
+#endif
```

Comment:
Source for the magic numbers: https://stackoverflow.com/questions/4113278/64-bit-multiplicative-hashing